xoto3/dynamodb/write_versioned/run.py

```diff
@@ -83,12 +83,12 @@ def versioned_transact_write_items(
 
     built_transaction = None
     for i, _ in enumerate(attempts_iterator or timed_retry()):
-        clean_transaction, built_transaction = lazy_batch_getting_transaction_builder(
+        built_transaction = lazy_batch_getting_transaction_builder(
             transaction_builder, item_keys_by_table_name, batch_get_item,
         )
-        if built_transaction is clean_transaction or _is_empty(built_transaction):
+        if _is_empty(built_transaction):
             logger.info("No effects were defined, so the existing items will be returned as-is.")
-            return clean_transaction
+            return built_transaction
```

Comment:
this change helps us make sure that presume has the semantics that you would expect - if you apply a presumed item value but no effect occurs, it's still nice to see that presumed item value in the returned transaction object, even though you didn't change it.

Reply:
:+1: